sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -8,150 +8,318 @@
 
 #pragma once
 
+#include <CL/sycl/detail/aligned_allocator.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/memory_manager.hpp>
+#include <CL/sycl/detail/scheduler/scheduler.h>
+#include <CL/sycl/detail/sycl_mem_obj.hpp>
+#include <CL/sycl/property_list.hpp>
+#include <CL/sycl/range.hpp>
+#include <CL/sycl/stl.hpp>
+
 namespace cl {
 namespace sycl {
 
-enum class image_channel_order : unsigned int {
-  a,
-  r,
-  rx,
-  rg,
-  rgx,
-  ra,
-  rgb,
-  rgbx,
-  rgba,
-  argb,
-  bgra,
-  intensity,
-  luminance,
-  abgr
-};
-
-enum class image_channel_type : unsigned int {
-  snorm_int8,
-  snorm_int16,
-  unorm_int8,
-  unorm_int16,
-  unorm_short_565,
-  unorm_short_555,
-  unorm_int_101010,
-  signed_int8,
-  signed_int16,
-  signed_int32,
-  unsigned_int8,
-  unsigned_int16,
-  unsigned_int32,
-  fp16,
-  fp32
-};
+enum class image_channel_order : unsigned int;
+enum class image_channel_type : unsigned int;
 
 namespace detail {
 
-template <int dimensions, typename AllocatorT> class image_impl {
-public:
-  image_impl(image_channel_order order, image_channel_type type,
-             const range<dimensions> &range,
-             const property_list &propList) {
-    assert(!"Not implemented");
+// utility functions and typedefs for image_impl
+using image_allocator = aligned_allocator<byte, /*alignment*/ 64>;
+
+// utility function: Returns the Number of Channels for a given Order.
+uint8_t getImageNumberChannels(image_channel_order Order);
+
+// utility function: Returns the number of bytes per image element
+uint8_t getImageElementSize(uint8_t NumChannels, image_channel_type Type);
+
+// validImageDataT: cl_int4, cl_uint4, cl_float4, cl_half4
+// To be used in get_access method. Uncomment after get_access is implemented.
+// template <typename T>
+// using is_validImageDataT = typename detail::is_contained<
+//    T, type_list<cl_int4, cl_uint4, cl_float4, cl_half4>>::type;
+
+template <int dimensions, typename AllocatorT = image_allocator>
```

Comment:
Could you rename dimensions -> Dimensions to align with other code.

Reply:
doing the change at all locations.